### PR TITLE
[mlir][nfc] Update vectorize-tensor-extract.mlir (2/N)

### PR DESCRIPTION
**Pull request by banach-space**
Tests in "vectorize-tensor-extract.mlir" are inconsistent and would
benefit from refactoring to:

* Clearly categorize tests into "contiguous load," "gather load," and
  "scalar load + broadcast" cases, reflecting the structure of
  tensor.extract vectorization.
* Unify variable naming (both MLIR and FileCheck).
* Ensure all tests exercise unmasked vectorization (masked vectorization
  is covered in "vectorize-tensor-extract-masked.mlir").
* Improve and standardize formatting.

These changes will make it easier to identify the test cases being
exercised and simplify future maintenance or refactoring.

This is patch 2/N in the series; below is a summary of the specific changes
in this patch.

----------------------------------------------------------------------
Summary of patch 2/N
----------------------------------------------------------------------

Makes all tests re-use the same TD sequence. This TD sequence has been
extracted to a deducted file:
  * "td/vectorize-with-patterns.mlir".

----------------------------------------------------------------------
 Previous patches:
----------------------------------------------------------------------

* https://github.com/llvm/llvm-project/pull/118977

### CODE REVIEW

**Review by hanhanW — APPROVED** [review]
The original check seems to be generated by a script.. Thanks for the improvements! I have a dumb question: what does column tensor mean?

**Comment by banach-space** [comment]
> The original check seems to be generated by a script.. Thanks for the improvements! 

Thanks for taking a look! Note that you have approved patch 2/N - shall I assume that you are OK with 1/N as well?

> I have a dumb question: what does column tensor mean?

That's a good question and you can help me validate my taxonomy. Let's assume that our row/column tensors have been extracted from a 2D tensor (`tensor<6x6xi32>`).

**Row tensor**: `tensor<1x6xi32>`,
```
. . . . . .
. . . . . .
x x x x x x
. . . . . .
. . . . . .
. . . . . .
```

**Column tensor:** `tensor<6x1xi32>`.
```
x . . . . .
x . . . . .
x . . . . .
x . . . . .
x . . . . .
x . . . . .
```

Now, the vectorizer will assume that the underlying memory access pattern is:
* a "contiguous load" for row tensor (all elements to read are adjacent in "memory"),
* a "gather load" for column tensor (because there's a "row" of elements between consecutive elements to read).

Admittedly, one big weakness of this approach is that tensors do not ... live in memory 😅 

Also, a "column tensor" could totally represent something that's contiguous in memory (e.g. when starting from e.g. `tensor<36x1xi32>` instead of `tensor<6x6xi32>`).

Hope this makes sense :)

**Review by hanhanW — CHANGES_REQUESTED** [review]
Yes, it makes sense to me. And wait! I clicked a wrong commit, so what I reviewed is the (1/N) commit. Let me approve the other one, and take a look at this one.

File: mlir/test/Dialect/Linalg/vectorize-tensor-extract.mlir [hunk 1]
```diff
@@ -92,51 +76,35 @@ func.func @extract_scalar_from_0d_into_1d(%src: tensor<f32>, %init: tensor<1xf32
...
-    transform.structured.vectorize %0 { vectorize_nd_extract }  : !transform.any_op
-    transform.yield
-   }
-}
+// CHECK-SAME:      %[[SRC:.*]]: tensor<3x3xf32>,
+// CHECK-SAME:      %[[INIT:.*]]: tensor<1x1x3xf32>) -> tensor<1x1x3xf32> {
+// CHECK:           %[[C0:.*]] = arith.constant 0 : index
+// CHECK:           %[[C1:.*]] = arith.constant 1 : index
+// CHECK:           %[[C2:.*]] = arith.constant 2 : index
+// CHECK:           %[[PAD:.*]] = arith.constant 0.000000e+00 : f32
```
Comment (hanhanW) [inline mlir/test/Dialect/Linalg/vectorize-tensor-extract.mlir:104]: Why not using `CHECK-DAG` for constants? I remember that there was a commit could break the order. Using `CHECK-DAG` makes life easier if it happens again.

**Review by hanhanW — APPROVED** [review]
LG, thanks! Just a question about using `CHECK-DAG` or not.

File: mlir/test/Dialect/Linalg/vectorize-tensor-extract.mlir (hunk 1 above)
Comment (banach-space) [reply mlir/test/Dialect/Linalg/vectorize-tensor-extract.mlir:104]: We should absolutely use `CHECK-DAG` :) Sending fix shortly!

**Review by dcaballe — APPROVED** [review]

**Pull request merged**